code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/internal/MessagingCacheUtilities.java

```diff
@@ -121,8 +124,12 @@ List<PropositionPayload> getCachedPropositions() {
      * @param propositionPayload the {@link List<PropositionPayload>} containing the message payload to be cached.
      */
     void cachePropositions(final List<PropositionPayload> propositionPayload) {
-        // clean any existing cached files first
-        clearCachedData();
+        // clean any existing cached files first if propositionPayload is null or empty
+        if (propositionPayload == null || propositionPayload.isEmpty()) {
+            clearCachedData();
+            return;
```

Comment:
should we be clearing cache in this case?

Reply:
nope, should be propositions only